src/main/java/org/apache/commons/lang3/math/NumberUtils.java

```diff
@@ -1776,7 +1773,7 @@ private static boolean withDecimalsParsing(final String str, final int beginIdx)
             if (decimalPoints > 1) {
                 return false;
             }
-            if (!isDecimalPoint && !Character.isDigit(str.charAt(i))) {
+            if (!isDecimalPoint && !Character.isDigit(str.charAt(i)) && !(i == str.length() - 1 && str.charAt(i) == '.')) {
```

Comment:
What happens in countries that flip the meaning of the comma and period?

Reply:
perhaps there could be another method that takes a Locale object. Otherwise, I don't think the current parser should concern itself with it

Reply:
Per current Javadoc "Parsable numbers include those Strings understood by Integer#parseInt(String), Long#parseLong(String), Float#parseFloat(String)} or Double#parseDouble(String)."

If those methods don't worry about countries that flip the meaning of the comma and period, we shouldn't do it here. That would be a breaking behavior change.

As to another method that uses Locales, that's in java.util so outside the advertised scope for commons-lang.